src/main/java/gomedic/logic/commands/EditCommand.java

```diff
@@ -1,105 +1,100 @@
 package gomedic.logic.commands;
 
-import static gomedic.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static gomedic.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static gomedic.logic.parser.CliSyntax.PREFIX_DEPARTMENT;
+import static gomedic.logic.parser.CliSyntax.PREFIX_ID;
 import static gomedic.logic.parser.CliSyntax.PREFIX_NAME;
 import static gomedic.logic.parser.CliSyntax.PREFIX_PHONE;
-import static gomedic.logic.parser.CliSyntax.PREFIX_TAG;
 import static gomedic.model.Model.PREDICATE_SHOW_ALL_ITEMS;
 import static java.util.Objects.requireNonNull;
 
-import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
-import java.util.Set;
 
 import gomedic.commons.core.Messages;
-import gomedic.commons.core.index.Index;
 import gomedic.commons.util.CollectionUtil;
 import gomedic.logic.commands.exceptions.CommandException;
 import gomedic.model.Model;
-import gomedic.model.commonfield.Address;
-import gomedic.model.commonfield.Email;
+import gomedic.model.commonfield.Id;
 import gomedic.model.commonfield.Name;
 import gomedic.model.commonfield.Phone;
-import gomedic.model.person.Person;
-import gomedic.model.tag.Tag;
+import gomedic.model.person.doctor.Department;
+import gomedic.model.person.doctor.Doctor;
+import gomedic.model.person.doctor.DoctorId;
 
 /**
- * Edits the details of an existing person in the address book.
+ * Edits the details of an existing doctor in the address book.
  */
 public class EditCommand extends Command {
```

Comment:
Yup also just change this to `edit t/doctor` instead of using doctor as placeholder, since you can edit the doctor anyway

Reply:
Hmm for this I think it could be better for me to do it in my `update doctor` issue pr instead? 

I think it would be better for us to merge the refactor changes first in case there is too many merge conflicts to resolve for the rest

Reply:
> > Some changes required
> 
> Made the changes to most of your suggestions, but as mentioned in one of my comments, should we merge the refactoring first? Since the `edit t/doctor` change is part of the `update doctor details` issue, maybe I can change that in another pr instead?

in my opinion, we can use this pr to resolve two issues at once
Otherwise other people might need to work with edit (which apparently becomes doctor) and might becomes conflict in the future actually but sure, I can merge this first, lmk your final decision